pkg/daemon/daemon.go

```diff
@@ -526,11 +526,11 @@ func (dn *Daemon) isDesiredMachineState() (bool, string, error) {
 	// config, the machine is definitely not in its desired state. this function
 	// will return true (meaning they are reconcilable) if there aren't actually
 	// changes.
-	reconcilable, err := dn.reconcilable(currentConfig, desiredConfig)
+	reconcilableError, err := dn.reconcilable(currentConfig, desiredConfig)
 	if err != nil {
 		return false, "", err
 	}
-	if !reconcilable {
+	if reconcilableError != nil {
```

Comment:
Since we record the event and log to journal down in `update`, we should perhaps do it here as well

Reply:
Hum...I started to refactor all of this but...what we do if reconciliation fails there in `isDesiredMachineState()` is then later call down to the core `update()` function again, and we'll log there.  Right?

Reply:
Ah, you are correct, I forgot we removed the degradation for non-reconcilbale states, apologies for the noise